File: dsa-problems/NumberTheory/solutions/python/NUM-010-sum-divisors-range.py
import sys

def range_sigma(L: int, R: int) -> int:
    MOD = 1000000007
    sigma = [0] * (R + 1)
    
    for i in range(1, R + 1):
        for j in range(i, R + 1, i):
            sigma[j] += i
            
    total = 0
    for i in range(L, R + 1):
        total = (total + sigma[i]) % MOD
        
    return total

def main():
    input = sys.stdin.read
    data = input().split()
    if not data:
        return
    L = int(data[0])
    R = int(data[1])
    print(range_sigma(L, R))

if __name__ == "__main__":
    main()

File: dsa-problems/NumberTheory/solutions/javascript/NUM-010-sum-divisors-range.js
const readline = require("readline");

function rangeSigma(L, R) {
  const MOD = 1000000007n;
  const sigma = new BigInt64Array(R + 1);
  
  for (let i = 1; i <= R; i++) {
    const val = BigInt(i);
    for (let j = i; j <= R; j += i) {
      sigma[j] += val;
    }
  }
  
  let total = 0n;
  for (let i = L; i <= R; i++) {
    total = (total + sigma[i]) % MOD;
  }
  
  return total.toString();
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let data = [];
rl.on("line", (line) => data.push(...line.trim().split(/\s+/)));
rl.on("close", () => {
  if (data.length === 0) return;
  const L = parseInt(data[0], 10);
  const R = parseInt(data[1], 10);
  console.log(rangeSigma(L, R));
});
